agent/bootstrap.go

```diff
@@ -864,7 +882,15 @@ func (b *Bootstrap) Start() error {
 			// gracefully handle repository renames
 			b.runCommand("git", "remote", "set-url", "origin", b.Repository)
 		} else {
-			b.runCommand("git", "clone", "-v", "--", b.Repository, ".")
+			// `git clone` can't accept options within quote like
+			// `git clone "-v --depth 1", so we need to split them
+			// by spaces and pass them individually.
+			gitCloneFlags := strings.Split(b.GitCloneFlags, " ")
+			gitCloneArguments := []string{"clone"}
+			gitCloneArguments = append(gitCloneArguments, gitCloneFlags...)
+			gitCloneArguments = append(gitCloneArguments, "--", b.Repository, ".")
```

Comment:
What about args/flags with spaces? e.g. I want to specify config during cloning:

```
GIT_CLONE_FLAGS="-v --config 'description = \'buildkite repo\''"
```

So we have a shell arg parser or something?